hudi-common/src/main/java/org/apache/hudi/common/table/view/HoodieTableFileSystemView.java

```diff
@@ -299,6 +310,17 @@ void removeFileGroupsInPendingClustering(Stream<Pair<HoodieFileGroupId, HoodieIn
     });
   }
 
+  @Override
+  protected void removeFileGroupsInPendingClustering(String clusteringInstant) {
+    List<HoodieFileGroupId> fileGroupIdsToRemove = fgIdToPendingClustering.entrySet()
+        .stream().filter(fgIdInstantEntry -> clusteringInstant.equals(fgIdInstantEntry.getValue().getTimestamp()))
+        .map(Map.Entry::getKey).collect(Collectors.toList());
+    ValidationUtils.checkArgument(!fileGroupIdsToRemove.isEmpty(),
```

Comment:
Is this validation strictly necessary? If it's not found then also incremental sync can continue, isn't it?

Reply:
If we have arrived at this point, that means old timeline has replacecommit in-inflight and new timeline either completed it or rolled back. 
In that case, if we are not finding the filegroups to remove that means there maybe some other issues. So, it is could to fail incremental file sync and let full sync happen. If required at later point we could add monitoring for incremental sync to make sure everything is working as expected.